src/core/config/transformer/merge_types/type_merger.rs

```diff
@@ -351,7 +352,10 @@ mod test {
         config.types.insert("T1".to_string(), ty1);
         config.types.insert("T2".to_string(), ty2);
 
-        let config = TypeMerger::new(0.5).transform(config).to_result().unwrap();
+        let config = TypeMerger::new(0.5, false)
```

Comment:
Add a test to showcase the functionality when the flag is `true` because, in all examples, the flag is `false`.